src/v/datalake/datalake_manager.cc

```diff
@@ -336,8 +336,8 @@ datalake_manager::handle_translator_state_change(const model::ntp& ntp) {
     auto translation_ctx
       = translation::translation_context::make_default_translation_context(
         local_path{_writer_scratch_space},
-        partition->ntp(),
-        partition->get_revision_id(),
+        data_src->ntp(),
+        data_src->topic_revision(),
```

Comment:
>should track topic_revision_id
>Rather than raft group revision.

This makes sense. Did this surface as a test failure?

Reply:
yeah. the integration test mentioned here: https://github.com/redpanda-data/redpanda/pull/25152#issuecomment-2693035265